api/controllers/UserController.js

```diff
@@ -100,8 +96,27 @@ module.exports = {
   },
 
   logout: function (req, res) {
-    delete req.session.usr;
-    req.session.loggedIn = false;
-    return res.ok();
+    req.session.destroy(function afterDestroy() {
+      return res.ok();
+    });
+  },
+
+  status: async function (req, res) {
+    const { usr: id, loggedIn: authenticated } = req.session;
+    if (!authenticated || !id) {
+      return res.ok({
+        authenticated,
+      });
+    }
+    try {
+      const { username } = await userAPI.findUser(id);
+      return res.ok({
+        id,
+        username,
+        authenticated,
+      });
+    } catch {
+      return res.badRequest('Unable to get user status');
+    }
```

Comment:
New method to get a user's status when we bootstrap Vue, after we create the store but prior to creating the app. It will send back the data we need for the authenticated status.